### PR TITLE
Refactor action count stat computation

### PR DESCRIPTION
**Pull request by ncondron42**
- Fixes issue: identical actions immediately after one another are only counted once (ex: spotdodge > spotdodge with no shield/wait frames in between)
- Fixes issue: Not counting dash grabs
- Fixes issue: Grab success not counted if frame perfect pummel/throw
- Adds feature: Adds action counts for standard attacks -- jabs, dash attack, tilts, smashes, and aerials
- Removed unnecessary boiler-plate for checking simple action states
- Added a few simple tests for attack counts
- Added tests for various edge cases -- boost grabs, frame perfect throws, identical actions, jab types, angled attacks (fsmash/ftilt), and GnW who is broken beyond belief

### CODE REVIEW

File: src/stats/actions.ts [hunk 1]
```diff
@@ -152,60 +142,92 @@ function handleActionCompute(state: PlayerActionState, indices: PlayerIndexedTyp
...
   // Grab last 3 frames
   const last3Frames = state.animations.slice(-3);
   const prevAnimation = last3Frames[last3Frames.length - 2] as number;
-  const newAnimation = currentAnimation !== prevAnimation;
+  const prevFrameCounter = state.actionFrameCounters[state.actionFrameCounters.length - 2] as number;
+
+  // New action if new animation or frame counter goes back down (repeated action)
+  const isNewAction =
+    currentAnimation !== prevAnimation ||
+    (currentAnimation === prevAnimation && prevFrameCounter > currentFrameCounter);
```
Comment (vinceau) [inline src/stats/actions.ts:156]:
```suggestion
  const isNewAction = currentAnimation !== prevAnimation || prevFrameCounter > currentFrameCounter;
```
Pretty sure if `currentAnimation !== prevAnimation` is false, then the next clause guarantees it to be true, thus you don't have to compare it twice.

**Review by vinceau — CHANGES_REQUESTED** [review]

File: src/stats/actions.ts [hunk 2]
```diff
@@ -152,60 +142,87 @@ function handleActionCompute(state: PlayerActionState, indices: PlayerIndexedTyp
...
 
   // Grab last 3 frames
   const last3Frames = state.animations.slice(-3);
   const prevAnimation = last3Frames[last3Frames.length - 2] as number;
-  const newAnimation = currentAnimation !== prevAnimation;
+  const prevFrameCounter = state.actionFrameCounters[state.actionFrameCounters.length - 2] as number;
+
+  // New action if new animation or frame counter goes back down (repeated action)
+  const isNewAction = currentAnimation !== prevAnimation || prevFrameCounter > currentFrameCounter;
+
```
Comment (vinceau) [inline src/stats/actions.ts:155]: nit: can we remove the newline on 155?

File: src/stats/actions.ts [hunk 3]
```diff
@@ -152,60 +142,87 @@ function handleActionCompute(state: PlayerActionState, indices: PlayerIndexedTyp
...
+  incrementCount("throwCount.back", currentAnimation === State.THROW_BACK);
 
-    incrementCount("wallTechCount.success", currentAnimation === State.WALL_TECH);
-    incrementCount("wallTechCount.fail", currentAnimation === State.MISSED_WALL_TECH);
+  // Techs
+  let opponentDir = 1;
+  if (playerFrame.positionX! > opponentFrame.positionX!) {
+    opponentDir = -1;
   }
+  const facingOpponent = playerFrame.facingDirection == opponentDir;
```
Comment (vinceau) [inline src/stats/actions.ts:216]:
```suggestion
  const facingOpponent = playerFrame.facingDirection === opponentDir;
```
Need a triple equals here.

File: src/stats/actions.ts [hunk 4]
```diff
@@ -152,60 +142,87 @@ function handleActionCompute(state: PlayerActionState, indices: PlayerIndexedTyp
...
+  incrementCount("throwCount.forward", currentAnimation === State.THROW_FORWARD);
+  incrementCount("throwCount.down", currentAnimation === State.THROW_DOWN);
+  incrementCount("throwCount.back", currentAnimation === State.THROW_BACK);
 
-    incrementCount("wallTechCount.success", currentAnimation === State.WALL_TECH);
-    incrementCount("wallTechCount.fail", currentAnimation === State.MISSED_WALL_TECH);
+  // Techs
+  let opponentDir = 1;
+  if (playerFrame.positionX! > opponentFrame.positionX!) {
+    opponentDir = -1;
```
Comment (vinceau) [inline src/stats/actions.ts:214]:
```suggestion
  const opponentDir = (playerFrame.positionX! > opponentFrame.positionX!)  ? -1 : 1;
```
nit: can we use a ternary here?

**Review by vinceau — CHANGES_REQUESTED** [review]

**Review by vinceau — APPROVED** [review]
LGTM
